File: Chapter 1/Bonus Problems/reverseWords.js
// Given a string s, reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order.


// Example 1:

var s = "Let's take LeetCode contest"
// Output: "s'teL ekat edoCteeL tsetnoc"


const reverseWords = (s) => { 
  return s.split(' ').map((word) => word.split('').reverse().join('')).join(' ')
}

console.log(reverseWords(s))

File: 2026/February-11-2026/reverse_words.py
# 557. Reverse Words in a String III
# Solved
# Easy
# Topics
# premium lock icon
# Companies
# Given a string s, reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order.

 

# Example 1:

# Input: s = "Let's take LeetCode contest"
# Output: "s'teL ekat edoCteeL tsetnoc"
# Example 2:

# Input: s = "Mr Ding"
# Output: "rM gniD"

s = "Let's take LeetCode contest"

def reverse_words(s: str) -> str: 
  return ' '.join(word[::-1] for word in s.split(' '))
  
print(reverse_words(s))